### PR TITLE
Fix vocabulary reload corruption caused by trailing newline handling in IndexLookup.load_assets

### PR DESCRIPTION
**Pull request by MalyalaKarthik66**
- Fixes an issue where trailing newlines in `vocabulary.txt` could introduce unintended empty tokens during asset reload.
- This ensures vocabulary (including special tokens) is reconstructed exactly as saved.
- Adds tests covering save/load consistency and trailing newline edge cases.

### CODE REVIEW

File: keras/src/layers/preprocessing/index_lookup_test.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+import tempfile
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/index_lookup_test.py]: Please use `self.get_temp_dir()` in the tests instead.
- you don't need a `with` so the code is not indented, which is nicer
- it will cleanup at the end of the test

File: keras/src/layers/preprocessing/index_lookup_test.py [hunk 2]
```diff
@@ -616,3 +617,99 @@ def test_ensure_known_vocab_size_without_vocabulary(self):
         ):
             input_data = ["sample", "data"]
             layer(input_data)
+
+    def test_save_and_load_assets_string_vocab(self):
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/index_lookup_test.py:620]: Just for my understanding. This test was *not* failing before the fix you added in `index_lookup.py`, right? This is just adding coverage.

File: keras/src/layers/preprocessing/index_lookup_test.py [hunk 3]
```diff
@@ -616,3 +617,99 @@ def test_ensure_known_vocab_size_without_vocabulary(self):
...
+            )
+
+            self.assertEqual(vocab_before, vocab_after)
+            self.assertEqual(vocab_before_no_special, vocab_after_no_special)
+
+            # Verify outputs match
+            output_after = layer2(sample_input).numpy()
+            np.testing.assert_array_equal(output_before, output_after)
+
+    def test_save_and_load_assets_with_multiple_oov_indices(self):
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/index_lookup_test.py:659]: Just for my understanding. This test was *not* failing before the fix you added in `index_lookup.py`, right? This is just adding coverage.

**Review by hertschuh — COMMENTED** [review]
Thank you for the fix!

File: keras/src/layers/preprocessing/index_lookup_test.py (hunk 1 above)
Comment (MalyalaKarthik66) [reply keras/src/layers/preprocessing/index_lookup_test.py]: Updated to use `self.get_temp_dir()`.

File: keras/src/layers/preprocessing/index_lookup_test.py (hunk 2 above)
Comment (MalyalaKarthik66) [reply keras/src/layers/preprocessing/index_lookup_test.py:620]: Yes, these add coverage to ensure save/load behavior (including special tokens) remains stable and prevents regressions.
